poweractions/poweractions.py

```diff
@@ -0,0 +1,231 @@
+                               "https://ss14.io/watchdog, http://localhost:1212)")
+                return
+
+            cur_servers[view.modal.name.value] = {
+                "address": view.modal.url.value,
+                "key": view.modal.key.value,
+                "token": view.modal.token.value
+            }
+
+        await ctx.send("Server added successfully.")
+
+    @poweractionscfg.command()
+    async def remove(self, ctx: commands.Context, name: str) -> None:
+        """
+        Removes a server.
+
+        `<name>`: The name of the server to remove.
+        """
+        async with self.config.guild(ctx.guild).servers() as cur_servers:
+            if name not in cur_servers:
+                await ctx.send("That server did not exist.")
+                return
+
+            del cur_servers[name]
+
+        await ctx.tick()
+
+    @poweractionscfg.command()
+    async def list(self, ctx: commands.Context) -> None:
+        """
+        Get a list of servers.
+        """
+        servers = await self.config.guild(ctx.guild).servers()
+
+        if len(servers) == 0:
+            await ctx.send("No servers are currently configured!")
+            return
+
+        content = "\n".join(map(lambda s: f"{s[0]}: `{s[1]['address']}`", servers.items()))
+
+        pages = list(pagify(content, page_length=1024))
+        embed_pages = []
+        for idx, page in enumerate(pages, start=1):
+            embed = discord.Embed(
+                title="Server List",
+                description=page,
+                colour=await ctx.embed_colour(),
+            )
+            embed.set_footer(text="Page {num}/{total}".format(num=idx, total=len(pages)))
+            embed_pages.append(embed)
+        await menus.menu(ctx, embed_pages, menus.DEFAULT_CONTROLS)
+
+    @checks.admin()
+    @commands.command()
+    async def restartserver(self, ctx: commands.Context, server) -> None:
+        """
+        Restarts a server.
+        """
+        async with ctx.typing():
+            selectedserver = await self.config.guild(ctx.guild).servers()
+
+            if server not in selectedserver:
+                await ctx.send("That server does not exist.")
+                return
+            else:
+                server = selectedserver[server]
+
+            authheader = "Basic " + base64.b64encode(f"{server['key']}:{server['token']}".encode("ASCII")).decode(
+                "ASCII")
+
+            try:
+                async with aiohttp.ClientSession() as session:
+                    async def load():
+                        async with session.post(server["address"] + f"/instances/{server['key']}/restart",
+                                                headers={"Authorization": authheader}) as resp:
+                            if resp.status != 200:
+                                await ctx.send(f"Failed to restart the server. Wrong status code: {resp.status}")
+                                return
```

Comment:
This should probably log the response contents so you can figure out what's wrong.